src/vodozemac/ecies.rs

```diff
@@ -0,0 +1,209 @@
+//! This module implements [ECIES](https://en.wikipedia.org/wiki/Integrated_Encryption_Scheme), the
+//! elliptic curve variant of the Integrated Encryption Scheme.
+//!
+//! Please take a look at the vodozemac documentation of this module for more
+//! info.
+
+#![allow(missing_debug_implementations)]
+use std::sync::{Arc, Mutex};
+
+use matrix_sdk_crypto::vodozemac::ecies;
+use wasm_bindgen::prelude::*;
+
+use super::Curve25519PublicKey;
+
+/// The result of an inbound ECIES channel establishment.
+#[wasm_bindgen(getter_with_clone)]
+pub struct InboundCreationResult {
+    /// The established ECIES channel.
+    pub channel: EstablishedEcies,
+    /// The plaintext of the initial message.
+    pub message: String,
+}
+
+/// The result of an outbound ECIES channel establishment.
+#[wasm_bindgen(getter_with_clone)]
+pub struct OutboundCreationResult {
+    /// The established ECIES channel.
+    pub channel: EstablishedEcies,
+    /// The initial encrypted message.
+    pub initial_message: String,
+}
+
+/// An unestablished ECIES session.
+#[wasm_bindgen]
+pub struct Ecies {
+    inner: Option<ecies::Ecies>,
+    public_key: Curve25519PublicKey,
+}
+
+#[wasm_bindgen]
+impl Ecies {
+    /// Create a new, random, unestablished ECIES session.
+    ///
+    /// This method will use the `MATRIX_QR_CODE_LOGIN` info.
```

Comment:
What is this constant referring to? Can we use an intra doc link to make it more explicit?

Reply:
The constant is referring to the constant used in vodozemac and defined in the MSC. We can't use an intra-doc link here. I'll expand the comment a bit, it makes less sense compared to the vodozemac one.

Reply:
Added here: 86f43b7f619aaf0d42e252bdb270db8bba22bd9d.